src/test/ui/const-generics/const-arg-in-const-arg.rs

```diff
@@ -0,0 +1,44 @@
+// revisions: min
+// FIXME(const_generics): This test currently causes an ICE because
+// we don't yet correctly deal with lifetimes, reenable this test once
+// this is fixed.
+#![cfg_attr(min, feature(min_const_generics))]
+
+const fn foo<T>() -> usize { std::mem::size_of::<T>() }
+const fn bar<const N: usize>() -> usize { N }
+const fn faz<'a>(_: &'a ()) -> usize { 13 }
+const fn baz<'a>(_: &'a ()) -> usize where &'a (): Sized { 13 }
+
+struct Foo<const N: usize>;
+fn test<'a, 'b, T, const N: usize>() where &'b (): Sized {
+    let _: [u8; foo::<T>()]; //~ ERROR generic parameters may not
+    let _: [u8; bar::<N>()]; //~ ERROR generic parameters may not
+    let _: [u8; faz::<'a>(&())]; //~ ERROR a non-static lifetime
+    let _: [u8; baz::<'a>(&())]; //~ ERROR a non-static lifetime
+    let _: [u8; faz::<'b>(&())]; //~ ERROR a non-static lifetime
+    let _: [u8; baz::<'b>(&())]; //~ ERROR a non-static lifetime
+
+    // NOTE: This can be a future compat warning instead of an error,
+    // so we stop compilation before emitting this error in this test.
+    let _ = [0; foo::<T>()];
+
+    let _ = [0; bar::<N>()]; //~ ERROR generic parameters may not
+    let _ = [0; faz::<'a>(&())]; //~ ERROR a non-static lifetime
+    let _ = [0; baz::<'a>(&())]; //~ ERROR a non-static lifetime
+    let _ = [0; faz::<'b>(&())]; //~ ERROR a non-static lifetime
+    let _ = [0; baz::<'b>(&())]; //~ ERROR a non-static lifetime
+    let _: Foo<{ foo::<T>() }>; //~ ERROR generic parameters may not
+    let _: Foo<{ bar::<N>() }>; //~ ERROR generic parameters may not
+    let _: Foo<{ faz::<'a>(&()) }>; //~ ERROR a non-static lifetime
+    let _: Foo<{ baz::<'a>(&()) }>; //~ ERROR a non-static lifetime
+    let _: Foo<{ faz::<'b>(&()) }>; //~ ERROR a non-static lifetime
+    let _: Foo<{ baz::<'b>(&()) }>; //~ ERROR a non-static lifetime
+    let _ = Foo::<{ foo::<T>() }>; //~ ERROR generic parameters may not
```

Comment:
looks like this causes both the github and the vs code highligher to break, what would be a good way to notify the relevant people here?

I personally don't think that this is blocking though.

Reply:
GitHub and Atom use https://github.com/zargony/atom-language-rust. VS Code's Rust syntax is [here](https://github.com/microsoft/vscode/blob/master/extensions/rust/syntaxes/rust.tmLanguage.json) It would be good to also check how it looks in [CodeMirror](https://github.com/codemirror/CodeMirror).